src/deepsparse/pipeline.py

```diff
@@ -133,6 +139,12 @@ def __init__(
         if engine_type.lower() == DEEPSPARSE_ENGINE:
             self._engine_args["scheduler"] = scheduler
 
+        self.context = context
+        if self.context and self.context.num_cores != num_cores:
```

Comment:
if num_cores is the default (aka `None`) let's not raise an error and just override num_cores to that of the context

Reply:
this does have the potential to miss unintended edge cases though where the user could intentionally pass in None for `num_cores` and something else for `Context.num_cores`.  Don't want to overengineer, can discuss solutions if you think this is an issue